src/CMake/SetUpThirdParty.cmake

```diff
@@ -77,6 +80,37 @@ FUNCTION(SET_UP_THIRD_PARTY pkg libdirextensions incdirextension libs)
         RETURN()
     ENDIF (NOT (EXISTS "${base_dir_val}"))
 
+
+    cmake_parse_arguments(PARSE_ARGV 1 sutp "" "LIBDIR;INCDIR" "LIBS")
+    if(NOT DEFINED sutp_LIBS)
+        if(IGNORE_THIRD_PARTY_LIB_PROBLEMS)
+            message(STATUS "  LIBS for ${pkg} have not been defined.")
+        else()
+            message(FATAL_ERROR "  LIBS for ${pkg} have not been defined.")
+        endif()
+    else()
+        set(libs ${sutp_LIBS})
+    endif()
+        
+
+    if(DEFINED sutp_LIBDIR)
+        set(libdir ${sutp_LIBDIR})
+    else()
+        # is this a better test?  It was used by our FindZlib.cmake
+        #  if (LINUX AND CMAKE_SIZEOF_VOID_P EQUAL 8 AND EXISTS ${base_dir_val}/lib64)
```

Comment:
Reason this is in draft -- the question of whether the more complex -if test should be used?

Reply:
In some of these cases the tpl's use a cmake helper that tells them `lib64`.  
But I don't think things like CMAKE_SIZEOF_VOID_P or even the cmake helper will be 100% reliable as a general solution. 

Most of the time we will find them in one place or the other -- should we just search both? If they are in both places, that would be an error or we would need more info.

Reply:
I think we've been getting by with checking for existence of `<lib-location>/lib64`. The only find module that didn't use that if-test was Zlib.  Not sure why.